src/core/thread/mle_ftd.cpp

```diff
@@ -1718,6 +1724,14 @@ void Mle::HandleTimeTick(void)
             }
         }
 
+        if (router.IsStateLinkRequest() && !mDelayedSender.HasAnyScheduledLinkRequest(router) &&
+            !router.IsWaitingForLinkAccept())
+        {
+            LogInfo("Router 0x%04x - Failed to schedule/send Link Request", router.GetRloc16());
+            RemoveNeighbor(router);
+            continue;
```

Comment:
Is `continue` useful here? The code below handling Router ID timeout seems independent of the neighbor state.

Reply:
I agree. I think we should not need the `continue` here. We also use `continue` in the next `if()` block for "Link Accept timeout" (which I guess we can also remove). I will remove both in this PR.

With the `continue`, the "route ID" `kMaxLeaderToRouterTimeout` (on Leader) will still be checked on the next tick call, so it should not really cause any problems (though it is unnecessary).